### DISCUSSION TITLE
bmp image loads very slow when reading from sdcard

### DISCUSSION THREAD

**Discussion by davidmonterocrespo24** [Bodmer/TFT_eSPI#2360] [Alerts]
If I save an image with a resolution of 320 x240 for example, it takes at least 4 seconds to finish displaying it on the screen.
Is there any way to make a buffer in the ram memory of the esp32?

When you load that same photo from an array it only takes less than a second, a major difference.

**Comment by Bodmer** [comment]
Yes, that is down to access speeds and the data format. The quickest way to display an image is to convert it to an array and store it in FLASH.  An example is provided:
https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Generic/TFT_Flash_Bitmap/TFT_Flash_Bitmap.ino

**Comment by Bodmer** [comment]
Another option is to use the SPIFFS or LittleFS filing system which allows files to be stored in the FLASH memory:

https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/TFT_SPIFFS_BMP

Other format images may load faster since the file size is smaller but they take longer to decode.
